05.ls/ls_command.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+require 'optparse'
+require 'etc'
+
+opt = ARGV.getopts('a', 'l', 'r')
+
+def filetype(ftype)
+  {
+    'file' => '-',
+    'directory' => 'd',
+    'characterSpecial' => 'c',
+    'blockSpecial' => 'b',
+    'fifo' => 'f',
+    'link' => 'l',
+    'socket' => 's'
+  }[ftype]
+end
+
+def permission(mode)
+  {
+    '0' => '---',
+    '1' => '--x',
+    '2' => '-w-',
+    '3' => '-wx',
+    '4' => 'r--',
+    '5' => 'r-x',
+    '6' => 'rw-',
+    '7' => 'rwx'
+  }[mode]
+end
+
+def l_option(file)
+  stat = File::Stat.new(file)
+  mode = stat.mode.to_s(8)
+  data = []
+  data << filetype(stat.ftype) + permission(mode[-3]) + permission(mode[-2]) + permission(mode[-1])
+  data << stat.nlink
+  data << Etc.getpwuid(stat.uid).name
+  data << Etc.getgrgid(stat.gid).name
+  data << stat.size
+  data << stat.mtime.strftime('%-m %e %k:%M')
+  data << file
+end
+
+if opt['a']
+  def file_names
+    Dir.glob('*', File::FNM_DOTMATCH).sort
+  end
+  if opt ['r']
+    def file_names
+      Dir.glob('*', File::FNM_DOTMATCH).sort.reverse
+    end
+  end
+  if opt['l']
+    file_names.each do |file|
+      puts l_option(file)[0] + ' ' + l_option(file)[1].to_s + ' ' + l_option(file)[2] + ' ' + l_option(file)[3] + ' ' + l_option(file)[4].to_s + ' ' + l_option(file)[5] + ' ' + l_option(file)[6]
```

Comment:
RuboCop に引っかかっていたのはここですかね。
たとえば以下のように書いたらどうでしょう？
```
data = []
data << l_option(file)[0]
data << l_option(file)[1]
data << l_option(file)[2]
data << l_option(file)[3]
data << l_option(file)[4]
data << l_option(file)[5]
data << l_option(file)[6]
puts data.join(" ")
```
あれ、似たような処理が繰り返されていますね……！
```suggestion
data = []
(0..6).each {|i| data << l_option(file)[i] }
puts data.join(" ")
```
のような感じで順を追って整理していく考え方もあります。
今回のように、同じような処理が繰り返されているところは「違う書き方ができるかもチャンス」なので、注目してみてください〜！